controlplane/kubeadm/controllers/upgrade.go

```diff
@@ -105,7 +106,10 @@ func (r *KubeadmControlPlaneReconciler) upgradeControlPlane(
 		return ctrl.Result{}, err
 	}
 
-	if status.Nodes <= *kcp.Spec.Replicas {
+	// RolloutStrategy is currently defaulted and validated to be RollingUpdate
+	// We can ignore MaxUnavailable because we are enforcing health checks before we get here.
+	maxNodes := *kcp.Spec.Replicas + int32(kcp.Spec.RolloutStrategy.RollingUpdate.MaxSurge.IntValue())
+	if status.Nodes < maxNodes {
```

Comment:
Is MaxUnavailable not used anywhere in the code?

Reply:
It is not used anywhere in kubeadmControlPlane controllers code.

Reply:
Is it used anywhere else?

Reply:
If not, we should probably remove it

Reply:
We are using MaxUnavailable in webhook code

Reply:
Hm, in terms of validation that makes sense, although it doesn't seem that the logic here actually makes use of it. We might want to move that field to MachineHealthCheck, and document how to orchestrate?

Reply:
I'm ok with it. Should we anyway keep validation in place here?

Reply:
@detiber @CecileRobertMichon thoughts?

Reply:
I guess in this implementation we don't need both MaxSurge _and_ MaxUnavailable since one is the opposite of the other. Makes sense to remove it and add it in the future if we actually need it to implement a more flexible rollout.

Reply:
If we do remove it we also need to update any docs/proposal that reference it

Reply:
If we do want to keep it for UX purposes, it should probably be a calculated field, under `Status`

Reply:
I am +1 for removing it complete. I agree with @CecileRobertMichon  that we can add it back in the future if necessary. If everyone agrees I remove MaxUnavailable and update the proposal?

Reply:
We should possibly remove it in this same PR, so the changes go in together. +1 from me as well.